### PR TITLE
feat: Use env var instead of tty for interactive sidecars

### PR DESCRIPTION
**Pull request by Vafilor**
**What this PR does**:

**Which issue(s) this PR fixes**:

Fixes onepanelio/core#

**Special notes for your reviewer**:

**Checklist**

Please check if applies

- [ ] I have added/updated relevant unit tests
- [ ] I have added/updated relevant documentation

Required 

- [x] I accept to release these changes under the Apache 2.0 License

### CODE REVIEW

File: pkg/workflow_execution.go [hunk 1]
```diff
@@ -466,7 +469,16 @@ func (c *Client) injectAccessForSidecars(namespace string, wf *wfv1.Workflow) ([
 		for si, s := range t.Sidecars {
 			//If TTY is true, sidecar needs to be accessible by HTTP
 			//Otherwise, we skip the sidecar
-			if s.TTY != true {
+
+			hasInjectIstio := false
+			for _, e := range s.Env {
+				if e.Name == "inject-istio" && e.Value == "true" {
```
Comment (rushtehrani) [inline pkg/workflow_execution.go:475]: Ideally we should have the env var indicate that this is an interactive sidecar... something like `ONEPANEL_INTERACTIVE_SIDECAR=true`

Comment (Vafilor) [reply pkg/workflow_execution.go:475]: @rushtehrani Good idea - on it.

File: db/yaml/maskrcnn/20201208155115.yaml [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+      workingDir: /mnt/src
+    nodeSelector:
+      beta.kubernetes.io/instance-type: '{{workflow.parameters.sys-node-pool}}'
+    sidecars:
+      - name: tensorboard
+        image: tensorflow/tensorflow:2.3.0
+        command: [sh, -c]
+        env:
+          - name: ONEPANEL_INTERACTIVE_SIDECAR
+            value: "true"
```
Comment (rushtehrani) [inline db/yaml/maskrcnn/20201208155115.yaml:133]: can we use single quotes for these?

Comment (Vafilor) [reply db/yaml/maskrcnn/20201208155115.yaml:133]: Sure, just a min.

File: db/go/20201208155115_replace_tty_with_env_var_for_tfod.go [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+	)
+}
+
+// Down20201208155115 rolls back the environment variable change
+func Down20201208155115(tx *sql.Tx) error {
+	// This code is executed when the migration is rolled back.
+	return updateWorkflowTemplateManifest(
+		filepath.Join("tfod", "20201130130433.yaml"),
+		tensorflowObjectDetectionWorkflowTemplateName,
+		map[string]string{
```
Comment (aleksandrmelnikov) [inline db/go/20201208155115_replace_tty_with_env_var_for_tfod.go:34]: @Vafilor Just to double check, is this Down migration intentional? It seems to add the same map string as the up string?

Comment (Vafilor) [reply db/go/20201208155115_replace_tty_with_env_var_for_tfod.go:34]: @aleksandrmelnikov Yes, those are the labels. The code will set the labels, so if we don't provide anything here it should wipe them out.

File: db/go/20201208155805_replace_tty_with_env_var_for_maskrcnn.go [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+			"used-by": "cvat",
+		},
+	)
+}
+
+// Down20201208155805 rolls back the environment variable change
+func Down20201208155805(tx *sql.Tx) error {
+	// This code is executed when the migration is rolled back.
+	return updateWorkflowTemplateManifest(
+		filepath.Join("maskrcnn", "20201115145814.yaml"),
```
Comment (aleksandrmelnikov) [inline db/go/20201208155805_replace_tty_with_env_var_for_maskrcnn.go:32]: Same as above, Down migration uses the same used-by?

File: db/go/20201208155115_replace_tty_with_env_var_for_tfod.go (hunk 3 above)
Comment (aleksandrmelnikov) [reply db/go/20201208155115_replace_tty_with_env_var_for_tfod.go:34]: Okay, that makes sense. But I see the same key: value values? For up and down?

**Review by aleksandrmelnikov — APPROVED** [review]

**Pull request merged**